src/axom/core/docs/sphinx/core_acceleration.rst

```diff
@@ -0,0 +1,51 @@
+.. ## Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+******************************************************
+Core acceleration
+******************************************************
+
+Axom Core provides support to ease the implementation of user applications that
+intend to support execution utilizing accelerators. In order to supply this,
+Axom provides a simple API to access memory and computation, using RAJA
+and Umpire to access accelerators.
+
+.. warning:: 
+   You do need to compile with RAJA and Umpire to fully leverage this Axom functionality. 
+   While both submodules will operate without these tools, they will offer little.
+
+The memory management submodule offers a singular interface which allows the user to leverage either C++ memory functions or Umpire, 
+depending on the availability of Umpire at compilation. It supports a simple operation set: allocate, deallocate, reallocate, and copy. 
+If Umpire is in use, an allocator can be specified -- see Umpire documentation(make this a link) for more details. Note that the fall-back to
+C++ memory functions is automatic, so one piece of code can handle standard C++ or C++ with Umpire. However, to use advanced features, such as 
+accessing unified memory, Umpire must be enabled, otherwise errors will occur at compilation.
+
+Here is an example of using Axom’s memory management tools:
+
+.. literalinclude:: ../../examples/core_acceleration.cpp
+   :start-after: _membasic_start
+   :end-before: _membasic_end
+   :language: C++
+
+Throughout Axom, acceleration is increasingly supported. Both internally, and to support users, Axom Core offers an 
+interface that, using RAJA and Umpire internally, provides easy access to for-loop level acceleration via the for-all model, 

```

Comment:
better stated if you replace "for-all model" with "parallel-for idiom"

Reply:
replaced